### ISSUE TITLE
seg fault with azure cli authentication in azure-identity

### ISSUE THREAD

**Issue by davegaulke** [Azure/azure-sdk-for-python#11738]
- **Package Name**: azure-identity
- **Package Version**: 1.4.0 b3
- **Operating System**: Linux
- **Python Version**: 3.8.3

**Describe the bug**
I'm trying to use the new Azure CLI authentication integration and in a simple program that tries to get and set vault secrets, I get the segmentation fault given below.

```
Fatal Python error: Segmentation fault

Current thread 0x00007f0dcadda740 (most recent call first):
  File "/site-packages/azure/identity/_credentials/linux_vscode_adapter.py", line 73 in _get_refresh_token
  File "/site-packages/azure/identity/_credentials/linux_vscode_adapter.py", line 93 in get_credentials
  File "/site-packages/azure/identity/_credentials/vscode_credential.py", line 55 in get_token
  File "/site-packages/azure/identity/_credentials/chained.py", line 63 in get_token
  File "/site-packages/azure/identity/_credentials/default.py", line 121 in get_token
  File "/site-packages/azure/keyvault/secrets/_shared/challenge_auth_policy.py", line 137 in _handle_challenge
  File "/site-packages/azure/keyvault/secrets/_shared/challenge_auth_policy.py", line 111 in send
  File "/site-packages/azure/core/pipeline/policies/_retry.py", line 436 in send
  File "/site-packages/azure/core/pipeline/policies/_redirect.py", line 157 in send
  File "/site-packages/azure/core/pipeline/_base.py", line 71 in send
  File "/site-packages/azure/core/pipeline/_base.py", line 71 in send
  File "/site-packages/azure/core/pipeline/_base.py", line 71 in send
  File "/site-packages/azure/core/pipeline/_base.py", line 71 in send
  File "/site-packages/azure/core/pipeline/_base.py", line 208 in run
  File "/site-packages/azure/keyvault/secrets/_shared/_generated/v7_1_preview/operations/_key_vault_client_operations.py", line 3200 in set_secret                                                                             
  File "/site-packages/azure/keyvault/secrets/_client.py", line 112 in set_secret
  File "/site-packages/azure/core/tracing/decorator.py", line 83 in wrapper_use_tracer
  File "vault.py", line 18 in set_secret
```

**Comment by kaerm** [comment]
Thanks @davegaulke for reporting this, @chlowell can you take a look at this

**Comment by xiangyan99** [comment]
Could you send us your re-pro steps?

I tried on Ubuntu 18.04 with Python 3.8.3.

I ran the code:

from azure.identity._credentials.vscode_credential import VSCodeCredential

def test_live():
    credential = VSCodeCredential()
    str=credential.get_token('https://vault.azure.net/.default')
    print(str)

It works.

**Comment by davegaulke** [comment]
I'm running Python 3.8.3 on Clear Linux.
I did this:
```

Python 3.8.3 (default, May 14 2020, 13:31:01) 
[GCC 10.1.1 20200513 releases/gcc-10.1.0-36-gf2b77b928a] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from azure.keyvault.secrets import SecretClient
>>> from azure.identity import DefaultAzureCredential
>>> 
>>> KVUri = 'https://<vault_name>.vault.azure.net/'
>>> credential = DefaultAzureCredential()
>>> client = SecretClient(vault_url=KVUri, credential=credential)
>>> client.get_secret('some_key')
[1]    129722 segmentation fault (core dumped)  python
```

**Comment by xiangyan99** [comment]
Looks like there are some know compatibility issues for Python 3.8 & Clear Linux?

https://community.clearlinux.org/t/python-3-8-in-clear-linux-os/2371

**Comment by davegaulke** [comment]
Thanks for finding that. I was able to get it working in another environment, so I also suspect this is a Clear Linux python issue.

**Comment by davegaulke** [comment]
On further investigation, I believe this is not a Clear Linux issue. The key to reproducing this is to have libsecret installed. I have reproduced this in a python 3.8.3 container (debian buster based) with libsecret installed. Can you verify if libsecret was installed in your Ubuntu test? If not, please run again after installing libsecret.

**Comment by xiangyan99** [comment]
If LoadLibrary("libsecret-1.so.0") fails, we raise error not "segmentation fault". :)

I will double check

**Comment by davegaulke** [comment]
I would expect that the LoadLibrary succeeds when the segmentation fault happens, since it is the presence of the library that leads to the seg fault.

**Comment by xiangyan99** [comment]
I think so. So I assume when you saw "segmentation fault". this means you already had libsecret installed on Clear Linux?

**Comment by davegaulke** [comment]
Yes, libsecret was installed when seg fault happened. If you remove libsecret, it works.

**Comment by xiangyan99** [comment]
Yes. When libsecret is not installed, the vscode credential will be ignored. 

In the meantime, you can use credential = DefaultAzureCredential(exclude_visual_studio_code_credential=True) to disable the vscode credential.

And we will test Debian linux.

**Comment by xiangyan99** [comment]
Fixed in #11914. Will be available in July's release.
